pandas/core/reshape/merge.py

```diff
@@ -1292,8 +1292,11 @@ def _get_join_indexers(left_keys, right_keys, sort=False, how="inner", **kwargs)
     # factorize keys to a dense i8 space
     # `count` is the num. of unique keys
     # set(lkey) | set(rkey) == range(count)
-    lkey, rkey, count = fkeys(lkey, rkey)
 
+    if how == "right":
+        rkey, lkey, count = fkeys(rkey, lkey)
```

Comment:
Is there a more appropriate variable name to use here for the left hand sides? General approach looks good but I think variable reuse can be confusing for future debugging. OK with more appropriate names even if it increases diff